### PR TITLE
HDFS-17506. [FGL] Performance for phase 1

### PR DESCRIPTION
**Pull request by ZanderXu**
Add some benchmarks

### CODE REVIEW

**Comment by ferhui** [comment]
Thanks. What is the purpose of this PR? Provide a benchmark tool and each one can test the HDFS cluster performance? or just test the namenode performance locally?

**Comment by ZanderXu** [comment]
> Thanks. What is the purpose of this PR? Provide a benchmark tool and each one can test the HDFS cluster performance? or just test the namenode performance locally?

`FSNLockBenchmarkThroughput` is a tool, each one can do the performance tests through this tool after deploying FGL.
`TestFSNLockBenchmarkThroughput` is a UT, it mocks a `MiniQJMHACluster` and do some performance tests locally.

**Comment by ferhui** [comment]
> > Thanks. What is the purpose of this PR? Provide a benchmark tool and each one can test the HDFS cluster performance? or just test the namenode performance locally?
> 
> `FSNLockBenchmarkThroughput` is a tool, each one can do the performance tests through this tool after deploying FGL. `TestFSNLockBenchmarkThroughput` is a UT, it mocks a `MiniQJMHACluster` and do some performance tests locally.

Got it, thanks.

File: hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/namenode/fgl/FSNLockBenchmarkThroughput.java [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+      int readWriteRatio, int testingCount, int concurrency) {
+    this.fileSystem = fileSystem;
+    this.readWriteRatio = readWriteRatio;
+    this.testingCount = testingCount;
+    this.executorService = Executors.newFixedThreadPool(concurrency);
+  }
+
+  public void benchmark(String lockName) throws Exception {
+    System.out.println("Do benchmark for " + lockName);
+    Path basePath = new Path("/tmp/fsnlock/benchmark/throughput");
```
Comment (ferhui) [inline hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/namenode/fgl/FSNLockBenchmarkThroughput.java]: How about making the path as an input? if no input, can set a default path.

**Review by ferhui — COMMENTED** [review]
It's OK to me as an initial version. 
Just left minor comments.

**Comment by ferhui** [comment]
Can also add usage referring to hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/BenchmarkThroughput.java

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 22s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  |  The patch appears to include 2 new or modified test files.  |
|||| _ HDFS-17384 Compile Tests _ |
| -1 :x: |  mvninstall  |  35m 34s | [/branch-mvninstall-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6806/2/artifact/out/branch-mvninstall-root.txt) |  root in HDFS-17384 failed.  |
| +1 :green_heart: |  compile  |   0m 44s |  |  HDFS-17384 passed with JDK Ubuntu-11.0.22+7-post-Ubuntu-0ubuntu220.04.1  |
| +1 :green_heart: |  compile  |   0m 40s |  |  HDFS-17384 passed with JDK Private Build-1.8.0_402-8u402-ga-2ubuntu1~20.04-b06  |
| +1 :green_heart: |  checkstyle  |   0m 43s |  |  HDFS-17384 passed  |
| +1 :green_heart: |  mvnsite  |   0m 45s |  |  HDFS-17384 passed  |
| +1 :green_heart: |  javadoc  |   0m 43s |  |  HDFS-17384 passed with JDK Ubuntu-11.0.22+7-post-Ubuntu-0ubuntu220.04.1  |
| +1 :green_heart: |  javadoc  |   1m  9s |  |  HDFS-17384 passed with JDK Private Build-1.8.0_402-8u402-ga-2ubuntu1~20.04-b06  |
| +1 :green_heart: |  spotbugs  |   1m 56s |  |  HDFS-17384 passed  |
| +1 :green_heart: |  shadedclient  |  21m 58s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 37s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 38s |  |  the patch passed with JDK Ubuntu-11.0.22+7-post-Ubuntu-0ubuntu220.04.1  |
| +1 :green_heart: |  javac  |   0m 38s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 33s |  |  the patch passed with JDK Private Build-1.8.0_402-8u402-ga-2ubuntu1~20.04-b06  |
| +1 :green_heart: |  javac  |   0m 33s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| -0 :warning: |  checkstyle  |   0m 30s | [/results-checkstyle-hadoop-hdfs-project_hadoop-hdfs.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6806/2/artifact/out/results-checkstyle-hadoop-hdfs-project_hadoop-hdfs.txt) |  hadoop-hdfs-project/hadoop-hdfs: The patch generated 1 new + 0 unchanged - 0 fixed = 1 total (was 0)  |
| +1 :green_heart: |  mvnsite  |   0m 38s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   0m 32s |  |  the patch passed with JDK Ubuntu-11.0.22+7-post-Ubuntu-0ubuntu220.04.1  |
| +1 :green_heart: |  javadoc  |   1m  0s |  |  the patch passed with JDK Private Build-1.8.0_402-8u402-ga-2ubuntu1~20.04-b06  |
| +1 :green_heart: |  spotbugs  |   1m 43s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  |  21m 13s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| -1 :x: |  unit  | 212m 26s | [/patch-unit-hadoop-hdfs-project_hadoop-hdfs.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6806/2/artifact/out/patch-unit-hadoop-hdfs-project_hadoop-hdfs.txt) |  hadoop-hdfs in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 30s |  |  The patch does not generate ASF License warnings.  |
|  |   | 306m 48s |  |  |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hdfs.tools.TestDFSAdmin |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.45 ServerAPI=1.45 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6806/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/6806 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient spotbugs checkstyle codespell detsecrets |
| uname | Linux 3535f61db013 5.15.0-94-generic #104-Ubuntu SMP Tue Jan 9 15:25:40 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | HDFS-17384 / 3b26cc4461beb5090016838299add8be22d8dd97 |
| Default Java | Private Build-1.8.0_402-8u402-ga-2ubuntu1~20.04-b06 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.22+7-post-Ubuntu-0ubuntu220.04.1 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_402-8u402-ga-2ubuntu1~20.04-b06 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6806/2/testReport/ |
| Max. process+thread count | 4548 (vs. ulimit of 5500) |
| modules | C: hadoop-hdfs-project/hadoop-hdfs U: hadoop-hdfs-project/hadoop-hdfs |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6806/2/console |
| versions | git=2.25.1 maven=3.6.3 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m 01s |  |  No case conflicting files found.  |
| +0 :ok: |  spotbugs  |   0m 01s |  |  spotbugs executables are not available.  |
| +0 :ok: |  codespell  |   0m 01s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m 01s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m 00s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m 00s |  |  The patch appears to include 2 new or modified test files.  |
|||| _ HDFS-17384 Compile Tests _ |
| +1 :green_heart: |  mvninstall  | 128m 34s |  |  HDFS-17384 passed  |
| +1 :green_heart: |  compile  |  10m 30s |  |  HDFS-17384 passed  |
| +1 :green_heart: |  checkstyle  |   7m 37s |  |  HDFS-17384 passed  |
| +1 :green_heart: |  mvnsite  |  10m 54s |  |  HDFS-17384 passed  |
| +1 :green_heart: |  javadoc  |   9m 03s |  |  HDFS-17384 passed  |
| +1 :green_heart: |  shadedclient  | 218m 59s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   7m 15s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   5m 24s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   5m 24s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m 01s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  checkstyle  |   3m 41s |  |  the patch passed  |
| +1 :green_heart: |  mvnsite  |   6m 18s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   5m 27s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  | 224m 07s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   8m 28s |  |  The patch does not generate ASF License warnings.  |
|  |   | 614m 22s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| GITHUB PR | https://github.com/apache/hadoop/pull/6806 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient spotbugs checkstyle codespell detsecrets |
| uname | MINGW64_NT-10.0-17763 ab4e39a29189 3.4.10-87d57229.x86_64 2024-02-14 20:17 UTC x86_64 Msys |
| Build tool | maven |
| Personality | /c/hadoop/dev-support/bin/hadoop.sh |
| git revision | HDFS-17384 / 3b26cc4461beb5090016838299add8be22d8dd97 |
| Default Java | Azul Systems, Inc.-1.8.0_332-b09 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch-windows-10/job/PR-6806/1/testReport/ |
| modules | C: hadoop-hdfs-project/hadoop-hdfs U: hadoop-hdfs-project/hadoop-hdfs |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch-windows-10/job/PR-6806/1/console |
| versions | git=2.44.0.windows.1 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m 00s |  |  No case conflicting files found.  |
| +0 :ok: |  spotbugs  |   0m 01s |  |  spotbugs executables are not available.  |
| +0 :ok: |  codespell  |   0m 01s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m 01s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m 00s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m 00s |  |  The patch appears to include 2 new or modified test files.  |
|||| _ HDFS-17384 Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  88m 04s |  |  HDFS-17384 passed  |
| +1 :green_heart: |  compile  |   6m 18s |  |  HDFS-17384 passed  |
| +1 :green_heart: |  checkstyle  |   4m 39s |  |  HDFS-17384 passed  |
| +1 :green_heart: |  mvnsite  |   6m 24s |  |  HDFS-17384 passed  |
| +1 :green_heart: |  javadoc  |   5m 43s |  |  HDFS-17384 passed  |
| +1 :green_heart: |  shadedclient  | 144m 30s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   4m 44s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   3m 26s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   3m 26s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m 01s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  checkstyle  |   2m 17s |  |  the patch passed  |
| +1 :green_heart: |  mvnsite  |   3m 59s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   3m 26s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  | 155m 07s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   5m 14s |  |  The patch does not generate ASF License warnings.  |
|  |   | 415m 09s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| GITHUB PR | https://github.com/apache/hadoop/pull/6806 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient spotbugs checkstyle codespell detsecrets |
| uname | MINGW64_NT-10.0-17763 45a322de9117 3.4.10-87d57229.x86_64 2024-02-14 20:17 UTC x86_64 Msys |
| Build tool | maven |
| Personality | /c/hadoop/dev-support/bin/hadoop.sh |
| git revision | HDFS-17384 / 3b26cc4461beb5090016838299add8be22d8dd97 |
| Default Java | Azul Systems, Inc.-1.8.0_332-b09 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch-windows-10/job/PR-6806/2/testReport/ |
| modules | C: hadoop-hdfs-project/hadoop-hdfs U: hadoop-hdfs-project/hadoop-hdfs |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch-windows-10/job/PR-6806/2/console |
| versions | git=2.44.0.windows.1 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by ZanderXu** [comment]
The failed UT `hadoop.hdfs.tools.TestDFSAdmin` is not caused by this PR, I will create a ticket to fix it.

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 23s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  |  The patch appears to include 2 new or modified test files.  |
|||| _ HDFS-17384 Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  33m 21s |  |  HDFS-17384 passed  |
| +1 :green_heart: |  compile  |   0m 43s |  |  HDFS-17384 passed with JDK Ubuntu-11.0.22+7-post-Ubuntu-0ubuntu220.04.1  |
| +1 :green_heart: |  compile  |   0m 43s |  |  HDFS-17384 passed with JDK Private Build-1.8.0_402-8u402-ga-2ubuntu1~20.04-b06  |
| +1 :green_heart: |  checkstyle  |   0m 40s |  |  HDFS-17384 passed  |
| +1 :green_heart: |  mvnsite  |   0m 47s |  |  HDFS-17384 passed  |
| +1 :green_heart: |  javadoc  |   0m 44s |  |  HDFS-17384 passed with JDK Ubuntu-11.0.22+7-post-Ubuntu-0ubuntu220.04.1  |
| +1 :green_heart: |  javadoc  |   1m  7s |  |  HDFS-17384 passed with JDK Private Build-1.8.0_402-8u402-ga-2ubuntu1~20.04-b06  |
| +1 :green_heart: |  spotbugs  |   1m 53s |  |  HDFS-17384 passed  |
| +1 :green_heart: |  shadedclient  |  21m 56s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 38s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 42s |  |  the patch passed with JDK Ubuntu-11.0.22+7-post-Ubuntu-0ubuntu220.04.1  |
| +1 :green_heart: |  javac  |   0m 42s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 35s |  |  the patch passed with JDK Private Build-1.8.0_402-8u402-ga-2ubuntu1~20.04-b06  |
| +1 :green_heart: |  javac  |   0m 35s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  checkstyle  |   0m 31s |  |  the patch passed  |
| +1 :green_heart: |  mvnsite  |   0m 39s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   0m 30s |  |  the patch passed with JDK Ubuntu-11.0.22+7-post-Ubuntu-0ubuntu220.04.1  |
| +1 :green_heart: |  javadoc  |   1m  6s |  |  the patch passed with JDK Private Build-1.8.0_402-8u402-ga-2ubuntu1~20.04-b06  |
| +1 :green_heart: |  spotbugs  |   1m 51s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  |  21m 49s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  | 213m 38s |  |  hadoop-hdfs in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 31s |  |  The patch does not generate ASF License warnings.  |
|  |   | 306m 47s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.45 ServerAPI=1.45 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6806/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/6806 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient spotbugs checkstyle codespell detsecrets |
| uname | Linux 6412d6991c5d 5.15.0-94-generic #104-Ubuntu SMP Tue Jan 9 15:25:40 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | HDFS-17384 / db92d267a8d598d36f9fbe59ec8c24ccf754c558 |
| Default Java | Private Build-1.8.0_402-8u402-ga-2ubuntu1~20.04-b06 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.22+7-post-Ubuntu-0ubuntu220.04.1 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_402-8u402-ga-2ubuntu1~20.04-b06 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6806/3/testReport/ |
| Max. process+thread count | 4448 (vs. ulimit of 5500) |
| modules | C: hadoop-hdfs-project/hadoop-hdfs U: hadoop-hdfs-project/hadoop-hdfs |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6806/3/console |
| versions | git=2.25.1 maven=3.6.3 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

**Review by ferhui — APPROVED** [review]

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m 00s |  |  No case conflicting files found.  |
| +0 :ok: |  spotbugs  |   0m 01s |  |  spotbugs executables are not available.  |
| +0 :ok: |  codespell  |   0m 01s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m 01s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m 00s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m 01s |  |  The patch appears to include 2 new or modified test files.  |
|||| _ HDFS-17384 Compile Tests _ |
| +1 :green_heart: |  mvninstall  | 108m 02s |  |  HDFS-17384 passed  |
| +1 :green_heart: |  compile  |   7m 17s |  |  HDFS-17384 passed  |
| +1 :green_heart: |  checkstyle  |   5m 38s |  |  HDFS-17384 passed  |
| +1 :green_heart: |  mvnsite  |   8m 07s |  |  HDFS-17384 passed  |
| +1 :green_heart: |  javadoc  |   7m 02s |  |  HDFS-17384 passed  |
| +1 :green_heart: |  shadedclient  | 177m 17s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m 46s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   4m 29s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   4m 29s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m 00s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  checkstyle  |   2m 53s |  |  the patch passed  |
| +1 :green_heart: |  mvnsite  |   5m 08s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   4m 19s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  | 193m 49s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   6m 26s |  |  The patch does not generate ASF License warnings.  |
|  |   | 512m 36s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| GITHUB PR | https://github.com/apache/hadoop/pull/6806 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient spotbugs checkstyle codespell detsecrets |
| uname | MINGW64_NT-10.0-17763 a4a169f40776 3.4.10-87d57229.x86_64 2024-02-14 20:17 UTC x86_64 Msys |
| Build tool | maven |
| Personality | /c/hadoop/dev-support/bin/hadoop.sh |
| git revision | HDFS-17384 / db92d267a8d598d36f9fbe59ec8c24ccf754c558 |
| Default Java | Azul Systems, Inc.-1.8.0_332-b09 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch-windows-10/job/PR-6806/3/testReport/ |
| modules | C: hadoop-hdfs-project/hadoop-hdfs U: hadoop-hdfs-project/hadoop-hdfs |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch-windows-10/job/PR-6806/3/console |
| versions | git=2.44.0.windows.1 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by ferhui** [comment]
Thanks for contribution. Merged.

**Pull request merged**